enumeration/enumerator/resource_enumerator.go

```diff
@@ -0,0 +1,229 @@
+package enumerator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/snyk/driftctl/enumeration"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/snyk/driftctl/enumeration/alerter"
+	"github.com/snyk/driftctl/enumeration/parallel"
+	"github.com/snyk/driftctl/enumeration/remote"
+	"github.com/snyk/driftctl/enumeration/remote/common"
+	"github.com/snyk/driftctl/enumeration/resource"
+	"github.com/snyk/driftctl/enumeration/terraform"
+)
+
+type ResourceEnumerator struct {
+	alerter              *sliceAlerter
+	progress             enumeration.ProgressCounter
+	remoteLibrary        *common.RemoteLibrary
+	providerLibrary      *terraform.ProviderLibrary
+	enumeratorRunner     *parallel.ParallelRunner
+	detailsFetcherRunner *parallel.ParallelRunner
+	to                   string
+}
+
+func NewResourceEnumerator() *ResourceEnumerator {
+	return &ResourceEnumerator{
+		enumeratorRunner:     parallel.NewParallelRunner(context.TODO(), 10),
+		detailsFetcherRunner: parallel.NewParallelRunner(context.TODO(), 10),
+	}
+}
+
+func (e *ResourceEnumerator) Init(to, providerVersion, configDirectory string) error {
```

Comment:
If I understand correctly, for resource enumerator use cases (as opposed to drift detection), we still have some un-necessary parameters here. Suggestions:

`to`: can we infer this from the resource type? Or possibly as a stepping stone to avoid adding scope to this PR: remove the config type. "aws+tf" looks weird for resource enumeration, so how about just "aws"?

`providerVersion`: always "" for enumeration-only cases, we don't care about matching provider versions to a config

`configDirectory`: can we make "" automatically find a tempdir using `os.TempDir`?

We could do all of this with a sort of builder pattern:

```go
enumerator, err := NewResourceEnumerator().
  WithConfigDir(). // optional, "" finds a temp dir automatically
  WithCloud("aws").
  WithConfigType("tf"). // not for enumerator use-cases, but maybe for 
  Build()
```

In the driftctl-scan flow, meaningful parameters would be assigned, e.g. providerVersion from a flag if requested.

This is a pretty hasty sketch of mine, I haven't put a ton of thought into it, so don't take it as gospel!